common/stats/status_server.cpp

```diff
@@ -98,12 +101,33 @@ StatusServer::StatusServer(uint16_t port, EndPointToOPMap op_map)
 }
 
 std::string StatusServer::GetPageContent(const std::string& end_point) {
-  auto itor = op_map_.find(end_point);
-  if (itor == op_map_.end()) {
-    return "Unsupported http path.\n";
+  // Add dummy url to allow it to be parsed by folly:Uri.
+  folly::Uri u("http://blah.blah" + end_point);
+  auto params = u.getQueryParams();
+
+  std::vector<std::pair<std::string, std::string>> args;
+  if (!params.empty()) {
+    for (auto p : params) {
+      args.push_back(std::make_pair(p.first.toStdString(),
+                                    p.second.toStdString()));
+    }
+  }
+
+  auto iter = op_map_.find(u.path().toStdString());
+  if (iter == op_map_.end()) {
+    return  "Unsupported http path.\n";
+  }
+
+  std::string text_ext = ".txt";
+  if (u.path().substr(u.path().length() - text_ext.length(), text_ext.length())
+      == text_ext) {
+    return iter->second(&args).asString();
```

Comment:
This function can only be used to access ints and strings. For arrays, objects etc, the op will need to return a formatted string as a dynamic.